### PR TITLE
feat: Add native Docker integration to MCPM

### PR DESCRIPTION
**Pull request by chazmaniandinkle**
## Summary

This PR adds native Docker integration to MCPM, enabling seamless orchestration between MCP server profiles and Docker Compose services with bidirectional synchronization.

### 🐳 New `mcpm docker` Command Group
- `mcpm docker sync PROFILE_NAME` - Sync profile to Docker Compose
- `mcpm docker status` - Show Docker integration status  
- `mcpm docker deploy [SERVICES...]` - Deploy Docker services
- `mcpm docker generate PROFILE_NAME` - Generate Docker Compose from profile

### 🔄 Bidirectional Synchronization
- **Profile → Docker**: Generate Docker Compose services from MCPM profiles
- **Docker → Profile**: Create MCPM profiles from Docker Compose services
- **Conflict Resolution**: Intelligent handling when both sides change
- **Smart Detection**: Automatic server type detection and mapping

### 🏗️ Production-Ready Architecture
- **Service Detection**: Pattern-based server type identification
- **Network Segmentation**: Proper Docker networking with `mcp-network`
- **Health Checks**: Built-in container health monitoring
- **Volume Management**: Persistent data storage
- **Environment Variables**: Secure configuration management

### Core Components
- `DockerIntegration` class for Docker operations orchestration
- `DockerSyncOperations` class for bidirectional sync with conflict resolution
- Server mapping system for automatic Docker service generation
- Target operations integration for enhanced workflow support

### Backward Compatibility
- ✅ 100% backward compatible with existing MCPM functionality  
- ✅ Docker integration is entirely opt-in
- ✅ Existing profiles and commands work unchanged
- ✅ No breaking changes to existing workflows

## Testing
- Comprehensive test suite with 21 passing tests
- Unit tests for individual components
- Integration tests for end-to-end workflows
- Error handling and edge case validation

## Documentation
- Complete RFC document outlining architecture and design decisions
- Comprehensive inline code documentation
- Test examples demonstrating usage patterns

## Value Proposition
This transforms MCPM from a client-side tool into a full-stack MCP orchestration platform, enabling:
- **Production Deployments**: Container orchestration with proper monitoring
- **Development Workflow**: Seamless transition from local to containerized
- **Enterprise Readiness**: Scalable, isolated, and monitored MCP deployments
- **Community Benefit**: Docker deployment capabilities for all MCPM users

The implementation maintains MCPM's core strengths while adding powerful containerization capabilities that bridge the gap between development and production.

___

### **PR Type**
Enhancement

___

### **Description**
- Add native Docker integration with `mcpm docker` command group

- Implement bidirectional sync between profiles and Docker Compose

- Support PostgreSQL, Context7, GitHub, Obsidian server types

- Include comprehensive test suite with 21 passing tests

___

### **Changes diagram**

```mermaid
flowchart LR
  A["MCPM Profiles"] -- "sync" --> B["Docker Compose"]
  B -- "deploy" --> C["Docker Services"]
  B -- "reverse sync" --> A
  D["CLI Commands"] --> E["mcpm docker sync/status/deploy/generate"]
  E --> F["DockerIntegration Class"]
  F --> G["DockerSyncOperations Class"]
```

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td><strong>cli.py</strong><dd><code>Add docker command to CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/pathintegral-institute/mcpm.sh/pull/204/files#diff-b18da6768b002e1b119c978937265dfdbf26a54ed7d451da7c4e2ab77605bf4d">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>docker.py</strong><dd><code>Implement Docker integration commands</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/pathintegral-institute/mcpm.sh/pull/204/files#diff-27612cb187c4212e9053cfef7f7296855cc53772b3736f207a8f6c9db0affa53">+371/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>docker_sync.py</strong><dd><code>Add bidirectional Docker sync operations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/pathintegral-institute/mcpm.sh/pull/204/files#diff-53bb8ddbad357990472ac574a49645164a352e1deb4b645da7986425e23d4144">+362/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>test_docker_integration.py</strong><dd><code>Add comprehensive Docker integration tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/pathintegral-institute/mcpm.sh/pull/204/files#diff-eb308999f6144fc77d912466bb872bc5bc09d68b0ba0bc92e0af6cf2fccfdf56">+399/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test-docker-compose.yml</strong><dd><code>Add test Docker Compose configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/pathintegral-institute/mcpm.sh/pull/204/files#diff-fbeabe833059460141e64c94baeff9669ee8294cf7435ca1027f2304849a0c17">+46/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>.python-version</strong><dd><code>Set Python version to 3.11.11</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/pathintegral-institute/mcpm.sh/pull/204/files#diff-652bad7c1002badcf13f8de3a5096c2724ce54ae36cd68a21ad280547dff8e87">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>RFC-DOCKER-INTEGRATION.md</strong><dd><code>Add Docker integration RFC document</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/pathintegral-institute/mcpm.sh/pull/204/files#diff-e97bfa371b3461ef318bc4a3c52a7c943fbcc50fec33ef577e07cb2d48374a15">+266/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

**Comment by chazmaniandinkle** [comment]
## 🛠️ Addressed Automated Review Findings

Updated the implementation to address all issues identified in the automated review:

### 🔒 **Security Improvements**
- Removed weak default credentials - now requires explicit environment variables
- Fixed Docker network connections to use service names instead of localhost  
- Added environment variable validation with user warnings

### 🐛 **Logic & Error Handling**
- Implemented proper change detection using file/profile hashing
- Added graceful JSON parsing error handling
- Enhanced subprocess error reporting with stderr output

### ✨ **CLI & UX**
- Added consistent help patterns with examples
- Improved error messages and user feedback
- Better validation warnings

### 🧪 **Testing**  
- Added 7 new test cases for security and error handling
- Total: 28 tests passing (up from 21)
- Enhanced coverage for edge cases

All changes maintain 100% backward compatibility.

File: src/mcpm/commands/docker.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+class DockerIntegration:
+    """Manages Docker integration for MCPM profiles."""
+    
+    def __init__(self, compose_file: str = "docker-compose.yml"):
+        self.compose_file = Path(compose_file)
+        self.profile_manager = ProfileConfigManager()
+        
+        # Server-to-Docker service mappings
+        self.server_mappings = {
```
[the remark this answers is not in the document]
Comment (chazmaniandinkle) [reply src/mcpm/commands/docker.py:36]: hmm, it should support anything that mcpm already allows, but it may have been implemented wrong. A lot of this logic is based on some wrappers I wrote around mcpm to synchronize it with a docker-compose mcp hub system, so some of that logic may have been lost in translation. I'll give it another look-over today.
